src/cargo/util/command_prelude.rs

```diff
@@ -1067,6 +1068,29 @@ fn get_targets_from_metadata() -> CargoResult<Vec<Target>> {
     Ok(targets)
 }
 
+pub fn get_dependency_name_candidates() -> Vec<clap_complete::CompletionCandidate> {
+    get_dependencies_from_metadata()
+        .unwrap_or_default()
+        .into_iter()
+        .map(|dep| clap_complete::CompletionCandidate::new(dep.name_in_toml().as_str()))
```

Comment:
I think this is supposed to be package name, not dependency name (`name_in_toml`)